tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -402,10 +402,17 @@ def run(self):
         except Exception:
             message = "Uncaught exception in TestRunnerManager.run:\n"
             message += traceback.format_exc()
-            self.logger.error(message)
+            self.logger.critical(message)
             raise
         finally:
             self.logger.debug("TestRunnerManager main loop terminating, starting cleanup")
+
+            while True:
+                _, _, test, _, _ = self.get_next_test()
+                if test is None:
+                    break
+                self.logger.critical(f"Test left in the queue: {test.id!r}")
```

Comment:
I might just log once rather than log each in a separate message (and maybe limit the message length since there could be tens of thousands of tests in the case something is really broken), but hopefully it doesn't come up that often.

Reply:
We normally log several times per test, so either way it's a lot in the logs.

We could also have a single CRITICAL message and then log the rest as ERROR.

Or change mozlog to be able to report NOTRUN for all the tests (because… ERROR seems wrong, they just haven't run)?

Reply:
I just think it obscures the actual error if you log all the tests. I'd prefer one `CRITICAL` message saying `"Tests left in the queue {first_test_id} and {n} more"`, but I'd also be OK with one critical level message saying `N tests left in the queue` and lower level messages for the actual tests.

Reply:
OK, I've done that.